### DISCUSSION TITLE
Passing in `initialState` does not merge with models initial state in next.js example

### DISCUSSION THREAD

**Discussion by xeoneux** [rematch/rematch#841] [Q&A] [answered]
In the rematch next.js example: https://github.com/vercel/next.js/blob/canary/examples/with-rematch/shared/store.js
While using rematch 2.0.0@next7

I'm passing in `initialState` with only some values that could be different.
Shouldn't the default behaviour be to merge the passed in initialState with the initialState defined in the models?

I have to do this to make it work:

```ts
const rematchInitialState = init<RootModel>({ models }).getState();

export const initStore = (initialState: any) => {
  return init<RootModel>({
    models,
    plugins: [immerPlugin()],
    redux: { initialState: { ...rematchInitialState, ...initialState } }, // Merging initial states
  });
};
```

Is this the correct approach?
Is there any configuration option to do this through the library itself?

**Comment by semoal** [comment, marked as answer]
We try to keep the redux section of init method as clean as possible. Some users probably doesn't want this behaviour by default, so it's easier to just do two spreads than creating a config to disable this behaviour.
